File: my_solution.js
class TreeNode {
    constructor(value = 0, leftChild = null, rightChild = null) {
        this.value = value
        this.left = leftChild
        this.right = rightChild
    }
}

function searchBT(root, target) { 
    if(!root) return false
    
    let queue = [root]
    while(queue.length){
        const curr = queue.shift()
        if(curr.value === target) return true
        if(curr.left) queue.push(curr.left)
        if(curr.right) queue.push(curr.right)        
    }
    return false
}

/*
        3
    29      4
  2       2.   9

> 3 29 4 2 2 9
*/

// Test Cases
var tree1 = new TreeNode(3, 
                         new TreeNode(29, 
                                      new TreeNode(2)), 
                         new TreeNode(4, null, 
                                      new TreeNode(2, 
                                                   new TreeNode(9))))
console.log(searchBT(null, 1)) // false
console.log(searchBT(tree1, 9)) // true
console.log(searchBT(tree1, 1)) // false
console.log(searchBT(tree1, 2)) // true
console.log(searchBT(new TreeNode(1), 2)) // false

File: my_solution.py
class TreeNode:
    def __init__(self, value = 0, left = None, right = None):
        self.value = value
        self.left = left
        self.right = right

def searchBT(root: TreeNode, target: int) -> bool:
    if not root:
        return False
    
    queue = [root]
    while len(queue):
        curr = queue.pop(0)
        if curr.value == target:
            return True
        if curr.left:
            queue.append(curr.left)
        if curr.right:
            queue.append(curr.right)

            
    return False

# Test Cases
tree1 = TreeNode(3, TreeNode(29, TreeNode(2)), TreeNode(4, None, TreeNode(2, TreeNode(9))))
print(searchBT(None, 1)) # False
print(searchBT(tree1, 9)) # True
print(searchBT(tree1, 1)) # False
print(searchBT(tree1, 2)) # True
print(searchBT(TreeNode(1), 2)) # False
